src/cleanfigure.m

```diff
@@ -864,25 +934,28 @@ function removeData(meta, handle, id_remove)
       return
   end
 
+  % Check whether this is a 3D plot
+  is3D = isAxis3D(meta.gca);
+
   % Extract the data from the current line handle.
   xData = get(handle, 'XData');
```

Comment:
Hm, so it is 17 vs 12 lines of code, so we would have won 5 lines. As much as i love encapsulation, i believe, that this is a bit too much.